### DISCUSSION TITLE
Feature request: Support clustered SQLite

### DISCUSSION THREAD

**Discussion by achetronic** [dani-garcia/vaultwarden#2789] [Ideas]
Hello there,
First of all, thank you for your amazing project 😄 

I have been thinking about a way to maintain an easy-to-use database on home environments which are a bit more powerful (kubernetes). The point is that SQLite is small and easy to maintain over the time. But, for having several replicas inside K8s, you should go for a more canonical DB like PostgreSQL, which adds a lot of overhead for a little environment.

What about adding support for rqlite or dqlite as database? They are basically SQLite, but HA with raft for consensus

**Comment by BlackDex** [comment]
If that backend is supported by Diesel we could support it. But as far as i know it isn't supported.

**Reply by achetronic** [reply]
By the moment, this is not supported so I could open a discussion for the feature there. Thank you! :)

**Comment by loderunner84** [comment]
Maybe fork with [dqlite](https://github.com/canonical/dqlite)?
